### PR TITLE
[v1.14] Add SPIRE connection to status

### PR DESCRIPTION
**Pull request by meyskens**
This is a backport of https://github.com/cilium/cilium/pull/26896 

The automatic backport failed in https://github.com/cilium/cilium/pull/27629 

```release-note
Add SPIRE connection to `cilium status`
```

```upstream-prs
$ for pr in 26896; do contrib/backporting/set-labels.py $pr done 1.14; done
```

### CODE REVIEW

**Comment by meyskens** [comment]
/test-backport-1.14

**Comment by meyskens** [comment]
/test-backport-1.14

**Comment by joestringer** [comment]
Backports performed the usual way will also have automatic release not correlation to the original PR. More specifically, it needs text like this in the PR description:

````
```upstream-prs
$ for pr in 26896; do contrib/backporting/set-labels.py $pr done 1.14; done
```
````

**Comment by joestringer** [comment]
> The automatic backport failed in https://github.com/cilium/cilium/pull/27629

Please consider detailing what these problems were for reviewers so that we can assess whether the differences between the PRs may materially affect the function of the PR now that it's backported, and so we can focus review on the risk areas for integration with older branches.

**Review by aanm — APPROVED** [review]

**Comment by tklauser** [comment]
/test-backport-1.14

**Review by tklauser — APPROVED** [review]

**Comment by meyskens** [comment]
/test-backport-1.14

File: api/v1/models/status_response.go
Comment (joestringer) [inline api/v1/models/status_response.go:1]: > This commit is a manual backport due to formatting and import conflicts
between main and v1.14 due to the removal of ipcache upstream.

What's this about ipcache removal? I don't see any mention of ipcache in the original commit, and I'm not aware of ipcache being "removed" in any way :thinking:

**Review by joestringer — APPROVED** [review]
Changes LGTM (I compared the original commit to this backported version side-by-side), but the commit message confused me. Is there something related to ipcache that we should be reviewing more closely?

File: api/v1/models/status_response.go
Comment (meyskens) [reply api/v1/models/status_response.go:1]: Let me try to re-word it, the ipcache was replaced with the nodemanager in the main branch which conflicted a lot of function signatures and imports

**Comment by meyskens** [comment]
/test-backport-1.14

**Comment by aditighag** [comment]
@meyskens Err... Looks like MLH added the read-to-merge label prematurely. Can you check if all the required tests passed? Please re-add the label once the PR is ready for merge.

**Comment by meyskens** [comment]
/test-backport-1.14

**Review by youngnick — APPROVED** [review]

**Pull request merged**